build.gradle.kts

```diff
@@ -15,7 +15,7 @@ dependencies {
   implementation("org.springframework.boot:spring-boot-starter-data-jpa")
   implementation("com.amazon.redshift:redshift-jdbc4-no-awssdk:1.2.45.1069")
   implementation("org.postgresql:postgresql:42.7.2")
-  implementation("uk.gov.justice.service.hmpps:hmpps-digital-prison-reporting-lib:3.7.4")
+  implementation("uk.gov.justice.service.hmpps:hmpps-digital-prison-reporting-lib:4.3.0")
```

Comment:
Unfortunately this would require more changes in order to work.
It would need to the Athena and Redshift clients to be created with the STS credentials.
These are the needed:
[Athena](https://github.com/ministryofjustice/hmpps-digital-prison-reporting-mi/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/configuration/AthenaConfig.kt)
[Redshift](https://github.com/ministryofjustice/hmpps-digital-prison-reporting-mi/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/configuration/RedshiftDataApiConfig.kt)
[STS](https://github.com/ministryofjustice/hmpps-digital-prison-reporting-mi/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/configuration/StsCredentialsProviderConfig.kt)

Reply:
The version can be updated but if the goal is for the async queries to work, this extra configuration would be needed.
And also in our helm values we would need to add a service account name similar [to the reporting mi application](https://github.com/ministryofjustice/hmpps-digital-prison-reporting-mi/blob/main/helm_deploy/values-dev.yaml#L5).

Reply:
Discussed with Stu and approving as the goal is simply to upgrade the library in this instance.